### PR TITLE
Allow template to be a whole directory

### PR DESCRIPTION
**Pull request by breml**
Use-case: process complete conf.d directories without the need to list every file separately.

### CODE REVIEW

**Comment by breml** [comment]
@jwilder just updated the README.

**Comment by breml** [comment]
@jwilder is there still something missing for this PR to be merged?

**Comment by jmahowald** [comment]
@breml  I also created https://github.com/jwilder/dockerize/pull/64.
Maybe we take my tests and use your implementation and readme?

**Comment by breml** [comment]
@jwilder could you please join and let us know what do you think about this PR and #64 and how you want us to precede.

**Comment by kramarz** [comment]
I am also loking for this feature simply because i have sites_enabled in my nginx, and everyone of them i need to transform into a template. Glad there is PR already.

**Comment by breml** [comment]
@jwilder would love to get some feedback.

**Comment by breml** [comment]
@jwilder I have updated the README, as you asked me to do. how do you want me to proceed?

File: dockerize.go [hunk 1]
```diff
@@ -230,7 +232,37 @@ func main() {
...
 			template, dest = parts[0], parts[1]
 		}
-		generateFile(template, dest)
+
+		fi, err := os.Stat(template)
+		if err != nil {
+			log.Fatalf("unable to stat %s, error: %s", template, err)
+		}
+		if fi.IsDir() {
+			if dest != "" {
```
Comment (jwilder) [inline dockerize.go]: Could you move this whole block L241-L262 to a separate func like `generateDir`?

File: dockerize.go [hunk 2]
```diff
@@ -230,7 +232,37 @@ func main() {
...
+			}
+
+			files, err := ioutil.ReadDir(template)
+			if err != nil {
+				log.Fatalf("bad directory: %s, error: %s", template, err)
+			}
+
+			for _, file := range files {
+				if dest == "" {
+					generateFile(template+string(os.PathSeparator)+file.Name(), "")
```
Comment (jwilder) [inline dockerize.go]: You can use `filepath.Join(template, file.Name)` here

File: dockerize.go [hunk 3]
```diff
@@ -230,7 +232,37 @@ func main() {
...
+				} else {
+					generateFile(template+string(os.PathSeparator)+file.Name(), dest+string(os.PathSeparator)+file.Name())
```
Comment (jwilder) [inline dockerize.go]: Ditto: `file path.Join(...)` for these.

**Comment by breml** [comment]
@jwilder just implemented your suggestions.

**Comment by jwilder** [comment]
Thanks @breml!

**Pull request merged**